libfwupdplugin/fu-smbios.c

```diff
@@ -863,13 +863,13 @@ fu_smbios_setup(FuSmbios *self, GError **error)
 	path = g_build_filename(sysfsfwdir, "dmi", "tables", NULL);
 	if (g_file_test(path, G_FILE_TEST_EXISTS)) {
 		g_autoptr(GError) error_local = NULL;
-		if (!fu_smbios_setup_from_path(self, path, &error_local)) {
-			if (!g_error_matches(error_local, G_FILE_ERROR, G_FILE_ERROR_ACCES)) {
-				g_propagate_error(error, g_steal_pointer(&error_local));
-				return FALSE;
-			}
-			g_debug("ignoring %s", error_local->message);
+		if (fu_smbios_setup_from_path(self, path, &error_local))
+			return TRUE;
```

Comment:
this makes my head hurt a bit. can we keep the ! return failure and just add a `} else { return TRUE;}`?

Reply:
sure